--- a/drivers/input/misc/qpnp-power-on.c	2022-02-26 19:04:57.000000000 +0800
+++ b/drivers/input/misc/qpnp-power-on.c	2023-01-22 20:43:55.151288195 +0800
@@ -1106,6 +1106,11 @@
 	}
 }
 
+static void panic_outdmesg(void)
+{
+	panic("A panic hot restart has been triggered");
+}
+
 static int qpnp_pon_input_dispatch(struct qpnp_pon *pon, u32 pon_type)
 {
 	int rc;
@@ -1182,6 +1187,10 @@
 		spin_unlock(&pon->fs_sync_lock);
 		schedule_delayed_work(&pon->fsync_timer, msecs_to_jiffies(4*FS_SYNC_INTERVAL_MS));
 	}
+
+	if (cfg->pon_type == PON_KPDPWR && key_status)
+		panic_outdmesg();
+
 	input_report_key(pon->pon_input, cfg->key_code, key_status);
 	input_sync(pon->pon_input);
 
